Assets/Tests/InputSystem/Plugins/EnhancedTouchTests.cs

```diff
@@ -1140,4 +1140,50 @@ public unsafe void EnhancedTouch_TouchSimulation_DisablesPointerDevicesWithoutDi
         Assert.That(Touchscreen.current.touches[0].isInProgress, Is.True);
         Assert.That(Touchscreen.current.touches[0].position.ReadValue(), Is.EqualTo(new Vector2(123, 234)));
     }
+
+    [Test]
+    [Category("EnhancedTouch")]
+    [TestCase(true)]
+    [TestCase(false)]
+    public void EnhancedTouch_ActiveTouchesGetCanceledOnFocusLoss_WithRunInBackgroundBeing(bool runInBackground)
+    {
+        runtime.runInBackground = runInBackground;
+
+        BeginTouch(1, new Vector2(123, 456));
+
+        Assert.That(Touch.activeTouches, Has.Count.EqualTo(1));
+        Assert.That(Touch.activeTouches[0].phase, Is.EqualTo(TouchPhase.Began));
+
+        runtime.PlayerFocusLost();
+
+        if (runInBackground)
+        {
+            // When running in the background, next update after focus loss sees touches cancelled
+            // and update after that sees them gone.
+
+            InputSystem.Update(InputUpdateType.Dynamic);
+
+            Assert.That(Touch.activeTouches, Has.Count.EqualTo(1));
```

Comment:
Move 2*Asserts + Update() + final Assert outside if-else if equal stimuli/behavior is required/expected to avoid duplicated code?

Reply:
By the way I liked this test case, easy to follow and understand what is expected and is going on.

Reply:
Done